aptos-move/framework/aptos-framework/sources/staking_contract.spec.move

```diff
@@ -352,17 +421,127 @@ spec aptos_framework::staking_contract {
         aborts_if seconds + config.recurring_lockup_duration_secs > MAX_U64;
         aborts_if old_locked_until_secs > new_locked_until_secs || old_locked_until_secs == new_locked_until_secs;
         aborts_if !exists<timestamp::CurrentTimeMicroseconds>(@aptos_framework);
+
+        let post post_store = global<Store>(staker);
+        let post post_staking_contract = simple_map::spec_get(post_store.staking_contracts, operator);
+        let post post_stake_pool = global<stake::StakePool>(post_staking_contract.owner_cap.pool_address);
+        ensures post_stake_pool.locked_until_secs == new_locked_until_secs;
+    }
+
+    spec schema CreateStakingContractWithCoinsAbortsif {
+        staker: signer;
+        operator: address;
+        voter: address;
+        amount: u64;
+        commission_percentage: u64;
+        contract_creation_seed: vector<u8>;
+
+        aborts_if commission_percentage < 0 || commission_percentage > 100;
+
+        aborts_if !exists<staking_config::StakingConfig>(@aptos_framework);
+        let config = global<staking_config::StakingConfig>(@aptos_framework);
+        let min_stake_required = config.minimum_stake;
+        aborts_if amount < min_stake_required;
+
+        // verify new_staking_contracts_holder()
+        include !exists<Store>(staker_address) ==> NewStakingContractsHolderAbortsIf;
+        let staker_address = signer::address_of(staker);
+        // property 1: The Store structure for the staker exists after the staking contract is created.
+        ensures exists<Store>(staker_address);
+
+        let store = global<Store>(staker_address);
+        let staking_contracts = store.staking_contracts;
+        aborts_if simple_map::spec_contains_key(staking_contracts, operator);
+
+        // verify create_stake_pool()
+        let seed_0 = bcs::to_bytes(staker_address);
+        let seed_1 = concat(concat(concat(seed_0, bcs::to_bytes(operator)), SALT), contract_creation_seed);
+        let resource_addr = account::spec_create_resource_address(staker_address, seed_1);
+        include CreateStakePoolAbortsIf {resource_addr};
+        aborts_if staker_address == resource_addr && !exists<Store>(staker_address)
+            && global<account::Account>(resource_addr).guid_creation_num + 21 >= account::MAX_GUID_CREATION_NUM;
+        ensures exists<account::Account>(resource_addr);
+        ensures exists<stake::StakePool>(resource_addr);
+
+        // verify stake::add_stake_with_cap()
+        include StakeAddStakeWithCapAbortsIf{ pool_address: resource_addr };
+
+        let post post_store = global<Store>(staker_address);
+        let post post_staking_contracts = post_store.staking_contracts;
+        // property 2: A staking contract is created and stored in a mapping within the Store resource.
+        ensures simple_map::spec_contains_key(post_staking_contracts, operator);
+    }
+
+    spec schema StakeAddStakeWithCapAbortsIf {
```

Comment:
As mentioned before, if a schema contains both aborts_if and ensures, make sure use a proper name.

Reply:
I think you have similar spec in the stake spec PR. Is it possible make it more modular by landing the stake module PR first and use the schema there?

Reply:
I'll try that later